libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFUserAccount.m

```diff
@@ -377,6 +385,42 @@ + (NSString *)loginStateDescriptionFromLoginState:(SFUserAccountLoginState)login
     return key;
 }
 
+- (BOOL)storeEncryptedPhoto:(NSData *)photoData path:(NSString *)photoPath error:(NSError **)error {
+    NSData *encryptedData = [self encryptPhoto:photoData];
+    if (!encryptedData) {
+        NSString *errorMessage = @"User photo data could not be encrypted.";
+        [SFSDKCoreLogger e:[self class] format:errorMessage];
+
+        if (error) {
+            *error = [NSError errorWithDomain:kSFSDKUserAccountManagerErrorDomain
+                                        code:SFSDKUserAccountManagerCannotEncrypt
+                                    userInfo:@{ NSLocalizedDescriptionKey : errorMessage }];
+        }
+        return NO;
+    }
+    
+    if (![encryptedData writeToFile:photoPath options:NSDataWritingAtomic error:error]) {
+        if (error) {
+            [SFSDKCoreLogger e:[self class] format:@"Unable to write photo to disk: %@", *error];
+        }
+        return NO;
+    }
+
+    return YES;
+}
+
+- (UIImage *)decryptPhoto:(NSString *)photoPath {
```

Comment:
autoCreate could be NO here - if the key does not exist and is created at this point, there is no chance it could decrypt the file.